src/os/layer/preset/preset.js

```diff
@@ -110,11 +113,65 @@ const updateDefault = function(layer, preset) {
   }
 };
 
+/**
+ * Get the selected Preset ID from saved Settings for the desired Layer
+ *
+ * @param {string} id The layer ID
+ * @return {?string}
+ */
+const getSavedPresetId = function(id) {
+  var lookup = /** @type {!Object<string, ?string>} */
+      (Settings.getInstance().get(SettingKey.SAVED_PRESET_IDS, {}));
+  return lookup[id];
+};
+
+/**
+ * Get the selected Preset ID from saved Settings for the desired Layer
+ *
+ * @param {string} id The layer ID
+ * @param {?string=} opt_presetId
+ */
+const setSavedPresetId = function(id, opt_presetId) {
+  var lookup = /** @type {!Object<string, ?string>} */
+      (Settings.getInstance().get(SettingKey.SAVED_PRESET_IDS, {}));
+  lookup[id] = opt_presetId || null;
+  Settings.getInstance().set(SettingKey.SAVED_PRESET_IDS, lookup);
+};
+
+/**
+ * Get the selected Preset ID from saved Settings for the desired Layer
+ *
+ * @param {string} id The layer ID
+ * @return {!boolean}
+ */
+const getSavedPresetClean = function(id) {
+  var lookup = /** @type {!Object<string, boolean>} */
+      (Settings.getInstance().get(SettingKey.SAVED_PRESET_CLEANS, {}));
+  return (lookup[id] === false) ? false : true;
+};
+
+/**
+ * Get the selected Preset ID from saved Settings for the desired Layer
+ *
+ * @param {string} id The layer ID
+ * @param {boolean} clean Is clean
+ */
+const setSavedPresetClean = function(id, clean) {
+  var lookup = /** @type {!Object<string, boolean>} */
+      (Settings.getInstance().get(SettingKey.SAVED_PRESET_CLEANS, {}));
+  lookup[id] = clean;
+  Settings.getInstance().set(SettingKey.SAVED_PRESET_CLEANS, lookup);
```

Comment:
Doing a comparison for the "clean" check I think would make sense on-init. Where possible I'll look for a "lastUpdated" date -- and, missing that, do a value-by-value comparison -- but only once for the session. That way we get the best of both worlds: speed during the session (just assume any change event means it's dirty) and accuracy when the system has desired push changes (assuming the User has not gone "Custom" -- since we don't change their settings once they set them manually).